lib/Builder.js

```diff
@@ -396,19 +395,19 @@ CompiledNode.prototype.generateHashes = function (nodes, fnOriginNames, idProvid
       .add(argHashes)
 
   if (this.literalPromise) {
-    hashBuilder.add(this._graph._oidProvider.getId(this.literalPromise))
+    hashBuilder.add(this._graph._valueHasher.getHash(this.literalPromise))
   }
 
   // compute the function hash, and mix that in.
-  var fnHash = this._graph._oidProvider.getId(this.fn)
+  var fnHash = this._graph._valueHasher.getHash(this.fn)
   var fnOriginName = fnOriginNames[fnHash]
   if (!fnOriginName) {
     fnOriginName = fnOriginNames[fnHash] = this.originalName
   }
   hashBuilder.add(fnOriginName)
 
   if (this.errorFn) {
-    hashBuilder.add(this._graph._oidProvider.getId(this.errorFn))
+    hashBuilder.add(this._graph._valueHasher.getHash(this.errorFn))
   }
 
   this.completeHash = hashBuilder.buildCompleteHash()
```

Comment:
do you happen to know how many bits this hash has? I think it's technically still possible to get collisions even with unique OIDs right? If it's wide enough probably not worth worrying about.

Reply:
I looks like 32 bits.